src/backend/db/feature-permissions.js

```diff
@@ -0,0 +1,219 @@
+const {google} = require('googleapis');
+const { OAuth2Client } = require('google-auth-library');
+
+/**
+ * Translate Group Keys to Group Ids
+ * 
+ * @param {*} db: The database interface object 
+ * @param {Array[String]} groupKeys 
+ * 
+ * @returns Array of Group Ids
+ */
+const keysToIds = async (db, groupKeys) => {
+  let groupIds = [];
+  if (groupKeys && groupKeys.length > 0) {
+    try {
+      let query = 'SELECT id FROM groups WHERE group_key IN (' + groupKeys.map(_ => '?').join(',') + ');'
+      groupIds = ( await db.raw(query, [...groupKeys] )).rows.map(row => row.id);
+    } catch(e) {
+      console.log(e);
```

Comment:
Minor: no need to address for this pull request, but I think in the future, we should return some error code here instead of swallowing the exception.